New proposal for Engine::Displayable

nmellado opened the pull request:
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.


* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Simplify API for `Engine::Displayable` and inheriting classes by using `Core::MultiIndexGeometry`.


* **What is the current behavior?** (You can also link to an open issue here)
Mesh types are statically typed on Core Mesh types. 


* **What is the new behavior (if this is a feature change)?**
Take benefit of `MultiIndexGeometry` to allow dynamic typing of the rendered objects: `Engine::GeometryDisplayable` will store a collection of geometry layers.
Expected API (right): 
![Image from iOS](https://user-images.githubusercontent.com/6310221/126151263-26e4a909-16c2-4279-b51c-38092eba13e2.jpg)



* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Will impact client code explicitly working with `Engine::Displayable` and inhering classes.


* **Other information**:

nmellado committed:
[engine] Prepare API of GeometryDisplayable

[engine] Remove unused class IndexedAttribArrayDisplayable

[engine] Prepare classes attributes (do not compile)

Code does not compile as only the attributes have been defined, but the
methods have not been updated.

[core] Mark GeometryIndexLayerBase::size() as const

[engine] define all virtual function for GeometryDisplayable (will be deleted).

When transition will be complete, these accessor might not be necessary.

[engine] GeometryDisplayable updateGL (not tested).

[engine] GeometryDisplayable works, for HelloRadium.

triangulate

nmellado committed:
[engine] Prepare API of GeometryDisplayable

[engine] Remove unused class IndexedAttribArrayDisplayable

[engine] Prepare classes attributes (do not compile)

Code does not compile as only the attributes have been defined, but the
methods have not been updated.

[core] Mark GeometryIndexLayerBase::size() as const

[engine] define all virtual function for GeometryDisplayable (will be deleted).

When transition will be complete, these accessor might not be necessary.

[engine] GeometryDisplayable updateGL (not tested).

[engine] GeometryDisplayable works, for HelloRadium.

triangulate

dlyr committed:
[engine] add GeometryDisplayableComponent

dlyr committed:
[engine] add GeometryDisplayableComponent

dlyr committed:
[tests] Use GeometryDisplayableComponent in HelloRadium (nothing drawn yet).

dlyr committed:
[tests] Use GeometryDisplayableComponent in HelloRadium (nothing drawn yet).

dlyr committed:
[core] Switch struct -> class for AbstractGeometry.

dlyr committed:
[core] Switch struct -> class for AbstractGeometry.

dlyr committed:
[core] Fix IndexedGeometry.

[core] Expose LayerKeyHash struct

Allow to reuse layers hash function outside of MultiIndexedGeometry

dlyr committed:
[core] Fix IndexedGeometry.

[core] Expose LayerKeyHash struct

Allow to reuse layers hash function outside of MultiIndexedGeometry

dlyr committed:
[core] Use macro for clone implementation.

debug type clone

make clone everywhere

dlyr committed:
[core] Use macro for clone implementation.

debug type clone

make clone everywhere

dlyr committed:
[engine] double wireframe (ad hoc for quad mesh).

clean code with using

forward renderer handle poly and quad layers.

dlyr committed:
[engine] double wireframe (ad hoc for quad mesh).

clean code with using

forward renderer handle poly and quad layers.

dlyr committed:
[core][engine] Use demangleType.

dlyr committed:
[core][engine] Use demangleType.

dlyr committed:
[engine] user BijectiveAssociation for translation table.

merge for GeometryDisplayable.

dlyr committed:
[engine] user BijectiveAssociation for translation table.

merge for GeometryDisplayable.

dlyr committed:
[core][engine] Move triangulate to core.

dlyr committed:
[core][engine] Move triangulate to core.

dlyr committed:
[engine] Add empty ctor.

dlyr committed:
[engine] Add empty ctor.

dlyr committed:
[engine] use geometrydisplayable for components.

dlyr committed:
[engine] use geometrydisplayable for components.

dlyr committed:
[engine] rename correspondance -> matching

[engine] mesh override getNumVertices for GeometryDisplayable.

remove uneeded export api on template

dlyr committed:
[engine] rename correspondance -> matching

[engine] mesh override getNumVertices for GeometryDisplayable.

remove uneeded export api on template

dlyr committed:
[engine] GeometrySystem use GeometryDisplayable comp.

dlyr committed:
[engine] GeometrySystem use GeometryDisplayable comp.

dlyr committed:
[examples] Start using GeometryDisplayable in draw prim.

dlyr committed:
[examples] Start using GeometryDisplayable in draw prim.

dlyr committed:
[engine] GeometryComponent clean include.

dlyr committed:
[engine] GeometryComponent clean include.

dlyr committed:
[engine] Renderer remove debug print.

remove forward old draw wireframe

dlyr committed:
[engine] Renderer remove debug print.

remove forward old draw wireframe

dlyr committed:
[engine] handle line index layer picking render mode.

[engine] Mesh remove debug print.

remove mesh debug message

 print mesh name on invalid layer

dlyr committed:
[engine] handle line index layer picking render mode.

[engine] Mesh remove debug print.

remove mesh debug message

 print mesh name on invalid layer

dlyr committed:
[core] MeshPrimitives remove debug print.

dlyr committed:
[core] MeshPrimitives remove debug print.

dlyr committed:
[core] IndexedGeometry remove debug print.

dlyr committed:
[core] IndexedGeometry remove debug print.

dlyr committed:
[examples] HelloRadium remove debug print.

dlyr committed:
[examples] HelloRadium remove debug print.

MathiasPaulin commented on src/Core/Geometry/MeshPrimitives.cpp:
```diff
@@ -90,6 +91,139 @@ TriangleMesh makeSharpBox( const Vector3& halfExts,
     return makeSharpBox( aabb, color, generateTexCoord );
 }
 
+MultiIndexedGeometry makeSharpBox2( const Aabb& aabb,
```
Do we need to have 2 methods to build sharpbox ? I think that this one, quad-based shar-box, should replace the triangulated one.

dlyr replied:
nope, will be done before merging of course.

MathiasPaulin commented on src/Engine/Scene/GeometryComponent.hpp:
```diff
@@ -98,10 +100,11 @@ class SurfaceMeshComponent : public GeometryComponent
-using PolyMeshComponent     = SurfaceMeshComponent<Ra::Core::Geometry::PolyMesh>;
+using TriangleMeshComponent        = SurfaceMeshComponent<Ra::Core::Geometry::MultiIndexedGeometry>;
+using GeometryDisplayableComponent = SurfaceMeshComponent<Ra::Core::Geometry::MultiIndexedGeometry>;
+using LineMeshComponent            = SurfaceMeshComponent<Ra::Core::Geometry::LineMesh>;
```
Why is this one not a `SurfaceMeshComponent<Ra::Core::Geometry::MultiIndexedGeometry>` also ?

dlyr replied:
I removed this delcaration, seems that it is not used.

dlyr commented on src/Engine/Data/Mesh.hpp:
```diff
@@ -47,6 +47,8 @@ class RA_ENGINE_API Vao
  * It maintains the attributes and keeps them in sync with the GPU.
  * \note Attribute names are used to automatic location binding when using shaders.
+ *
+ * \TODO Remove rendermode attribute
```
Maybe in a second round, now it is used by picking to determine picking render mode.
I'm not sure how to properly handle that.

dlyr committed:
[engine] Remove unused LineMeshComponent.

dlyr committed:
[engine] Remove unused LineMeshComponent.

dlyr committed:
[examples] add fixed light in skinning example.

dlyr committed:
[examples] add fixed light in skinning example.

dlyr committed:
[core] use inverse transpose for normal lbs.

dlyr committed:
[core] use inverse transpose for normal lbs.

dlyr committed:
[core] comment typo in trianglemesh

dlyr committed:
[core] comment typo in trianglemesh

dlyr committed:
[core][engine] use AttribArrayGeometry as skinning data.

tmp disable rotation center skinning

dlyr committed:
[core][engine] use AttribArrayGeometry as skinning data.

tmp disable rotation center skinning

dlyr committed:
[core] register rw on AttribArrayGeometry for SurfaceMeshComponent.

dlyr committed:
[core] register rw on AttribArrayGeometry for SurfaceMeshComponent.

dlyr commented:
note for futur skinning https://cims.nyu.edu/gcl/papers/EG2014-Lighting.pdf

dlyr committed:
[core] Temporary remove non working file from base code.

dlyr committed:
[core] Temporary remove non working file from base code.

dlyr committed:
[engine] fix geometry component (after rebase)

dlyr committed:
[engine] fix geometry component (after rebase)

dlyr committed:
[core] Remove dead code.

dlyr committed:
[core] Remove dead code.

dlyr committed:
[core] Add missing include for std_any.

dlyr committed:
[core] Add missing include for std_any.

dlyr committed:
[engine] Add setter for active layer.

dlyr committed:
[engine] Add setter for active layer.

dlyr committed:
[engine] Use GeometryDisplayable for DrawPrimitive::Triangle.

dlyr committed:
[engine] Use GeometryDisplayable for DrawPrimitive::Triangle.

dlyr committed:
[engine] Remove LineMesh (use GeometryDisplayable instead).

dlyr committed:
[engine] Remove LineMesh (use GeometryDisplayable instead).

dlyr committed:
[examples] Use GeometryDisplayable for Poly example.

dlyr committed:
[examples] Use GeometryDisplayable for Poly example.

dlyr committed:
[engine] Sphere use GeometryDisplayable.

dlyr committed:
[engine] Sphere use GeometryDisplayable.

dlyr committed:
[engine] Fix set render mode when set active key (to update picking render mode).

dlyr committed:
[engine] Fix set render mode when set active key (to update picking render mode).

dlyr committed:
[core][engine] Move grid to mesh primitives, small refactor also.

Plane/Quad uses QuadMesh.

Add ArrayOfLayerKeys to ease setup.

dlyr committed:
[core][engine] Move grid to mesh primitives, small refactor also.

Plane/Quad uses QuadMesh.

Add ArrayOfLayerKeys to ease setup.

dlyr committed:
[core] *Mesh are now using aliases to IndexedGeometry<>.

Factorize TopologicalMesh::to*Mesh.
Update topomesh unittest.

dlyr committed:
[core] *Mesh are now using aliases to IndexedGeometry<>.

Factorize TopologicalMesh::to*Mesh.
Update topomesh unittest.

dlyr committed:
[engine] Fix priority to rm triangles fill draw primitives.

dlyr committed:
[engine] Fix priority to rm triangles fill draw primitives.

dlyr committed:
[engine] Add createMeshFromGeometry2 for geometry displayable.

Also better wireframe line width.

dlyr committed:
[engine] Add createMeshFromGeometry2 for geometry displayable.

Also better wireframe line width.

dlyr committed:
[examples] create GeometryDisplayable instead of AttribArrayDisplayable for loaded logo.

dlyr committed:
[examples] create GeometryDisplayable instead of AttribArrayDisplayable for loaded logo.

dlyr committed:
[engine][core][gui] Use QuadMesh and GeometryDisplayable for Torus and Capsule.

dlyr committed:
[engine][core][gui] Use QuadMesh and GeometryDisplayable for Torus and Capsule.

dlyr committed:
[core][engine] Add FanOrStrip, Disk use GeometryDisplayable.

Add restart to layer data, and allow to draw triangle fan for instance.

dlyr committed:
[core][engine] Add FanOrStrip, Disk use GeometryDisplayable.

Add restart to layer data, and allow to draw triangle fan for instance.

dlyr committed:
[core] Factorize index layer code using macro.

dlyr committed:
[core] Factorize index layer code using macro.

dlyr committed:
[core] Add offset to append for GeometryIndexLayer.

dlyr committed:
[core] Add offset to append for GeometryIndexLayer.

dlyr committed:
[engine] Add triangle layer to disc (not only fan) for wireframe.

dlyr committed:
[engine] Add triangle layer to disc (not only fan) for wireframe.

dlyr committed:
[core] Typo in comment.

dlyr committed:
[core] Typo in comment.

dlyr committed:
[core] Cleanup and remove duplicate code in TopologicalMesh.

dlyr committed:
[core] Cleanup and remove duplicate code in TopologicalMesh.

dlyr committed:
[core][...] Rename TriangleMesh.?pp to AttribArayGeometry.?pp

Do not include IndexedGeometry in AttribArrayGeometry
Update includes path.

dlyr committed:
[core][...] Rename TriangleMesh.?pp to AttribArayGeometry.?pp

Do not include IndexedGeometry in AttribArrayGeometry
Update includes path.

dlyr committed:
[core] wip on attrib geom in skinning

dlyr committed:
[core] wip on attrib geom in skinning

dlyr committed:
[core] Add default layer key (init as invalid) to multi index geom.

Set default layer on deep copy.

dlyr committed:
[core] Add default layer key (init as invalid) to multi index geom.

Set default layer on deep copy.

dlyr committed:
[core] Switch skinnig reference mesth to multiindexgeometry.

dlyr committed:
[core] Switch skinnig reference mesth to multiindexgeometry.

dlyr committed:
[engine] Clean includes and remove commented code in Skinning.

dlyr committed:
[engine] Clean includes and remove commented code in Skinning.

dlyr committed:
[core] Refactor indexed geometry (comments, inline).

dlyr committed:
[core] Refactor indexed geometry (comments, inline).

dlyr committed:
[core] Simpler init of default key, reset when deep clear.

dlyr committed:
[core] Simpler init of default key, reset when deep clear.

dlyr committed:
[core] Clean and refactor RayCastTriangleMesh.

dlyr committed:
[core] Clean and refactor RayCastTriangleMesh.

dlyr committed:
[tests] Add MultiIndexGeometry default_layer_key test.

dlyr committed:
[tests] Add MultiIndexGeometry default_layer_key test.

dlyr committed:
[engine] Some autos and addAttrib for GeometryDisplayable.

dlyr committed:
[engine] Some autos and addAttrib for GeometryDisplayable.

dlyr committed:
[tests] more multi index test.

dlyr committed:
[tests] more multi index test.

dlyr committed:
[examples] Add strip to all primitives.

dlyr committed:
[examples] Add strip to all primitives.

dlyr committed:
[core] Add doc for makeGrid.

dlyr committed:
[core] Add doc for makeGrid.

dlyr committed:
[examples] Add textured quad strip to DrawPrimitives.

dlyr committed:
[examples] Add textured quad strip to DrawPrimitives.

dlyr committed:
[core] Refactor makePlaneGrid.

dlyr committed:
[core] Refactor makePlaneGrid.

dlyr committed:
[core] Add operator<< on std::set helper.

dlyr committed:
[core] Add operator<< on std::set helper.

dlyr committed:
[engine] QuadStrip use makePlaneGrid.

dlyr committed:
[engine] QuadStrip use makePlaneGrid.

dlyr committed:
[engine] Mesh output error if layer not added.

dlyr committed:
[engine] Mesh output error if layer not added.

dlyr committed:
[engine] Handle wireframe when there isn't any triangle layer.

dlyr committed:
[engine] Handle wireframe when there isn't any triangle layer.

dlyr committed:
[core][engine][example] Add aabb mesh primitive.

dlyr committed:
[core][engine][example] Add aabb mesh primitive.

dlyr committed:
[core] Obb is core Aabb instead of in class using.

dlyr committed:
[core] Obb is core Aabb instead of in class using.

dlyr committed:
[Engine] Do not draw wireframe if RO is already in line mode.

dlyr committed:
[Engine] Do not draw wireframe if RO is already in line mode.

dlyr committed:
[gui] Set pivot on alt right clic.

dlyr committed:
[gui] Set pivot on alt right clic.

dlyr committed:
[core] Add makeBox2, prepare transition.

dlyr committed:
[core] Add makeBox2, prepare transition.

dlyr committed:
[examples] DrawPrimitives add cube from makeBox2.

Also move first wireframe triangle out of filled triangle.

dlyr committed:
[examples] DrawPrimitives add cube from makeBox2.

Also move first wireframe triangle out of filled triangle.

dlyr committed:
[core] Template indexed geometry with layer type instead of index.

dlyr committed:
[core] Template indexed geometry with layer type instead of index.

dlyr committed:
[core] Add indices/set_indices templated function.

And use them in IndexedGeometry.
Clean deprecated code.
Move online to class definition.

dlyr committed:
[core] Add indices/set_indices templated function.

And use them in IndexedGeometry.
Clean deprecated code.
Move online to class definition.

dlyr committed:
[core] set_indices add layer if needed.

Also add some semantic helper functions.

dlyr committed:
[core] set_indices add layer if needed.

Also add some semantic helper functions.

dlyr committed:
[engine] Use MultiIndexedGeometry instead of old Mesh or LineMesh.

dlyr committed:
[engine] Use MultiIndexedGeometry instead of old Mesh or LineMesh.

dlyr committed:
[examples] DrawPrimitives, add spline and line strip.

dlyr committed:
[examples] DrawPrimitives, add spline and line strip.

dlyr committed:
[engine] Remove deprecated MeshPtr, LineMesh->MultiIndexedGeometry.

dlyr committed:
[engine] Remove deprecated MeshPtr, LineMesh->MultiIndexedGeometry.

dlyr committed:
[core] Remove explicit from MultiIndexedGeometry copy ctor.

dlyr committed:
[core] Remove explicit from MultiIndexedGeometry copy ctor.

dlyr committed:
[core] Remove explicit call to copy ctor.

dlyr committed:
[core] Remove explicit call to copy ctor.

dlyr committed:
[core] indices_with_lock return pair with layer_key to unlockLayer.

dlyr committed:
[core] indices_with_lock return pair with layer_key to unlockLayer.

dlyr committed:
[core] Everything (?) is MultiIndexed.

dlyr committed:
[core] Everything (?) is MultiIndexed.

dlyr committed:
[engine] Draw primitive auto (for TriangleMesh->MultiIndexed).

dlyr committed:
[engine] Draw primitive auto (for TriangleMesh->MultiIndexed).

dlyr committed:
[gui] Guizmo geom use MultiIndexed.

dlyr committed:
[gui] Guizmo geom use MultiIndexed.

dlyr committed:
[tests] Update tests according to Indexed->MultiIndexed.

dlyr committed:
[tests] Update tests according to Indexed->MultiIndexed.

dlyr committed:
[gui] Move gizmo todo to viewer (where it's instancied).

dlyr committed:
[gui] Move gizmo todo to viewer (where it's instancied).

dlyr committed:
[gui] Add todo to todo comment.

dlyr committed:
[gui] Add todo to todo comment.

dlyr committed:
[core] Fix MultiIndexGeometry append bad offset.

dlyr committed:
[core] Fix MultiIndexGeometry append bad offset.

dlyr committed:
[gizmo] Add empty line for readability.

dlyr committed:
[gizmo] Add empty line for readability.

dlyr committed:
[engine] Remove Mesh component alias, everything is SurfaceMesh.

dlyr committed:
[engine] Remove Mesh component alias, everything is SurfaceMesh.

dlyr committed:
[examples] Update according to SurfaceMeshComponent.

dlyr committed:
[examples] Update according to SurfaceMeshComponent.

dlyr committed:
[doc] Wip update mesh documentation.

dlyr committed:
[doc] Wip update mesh documentation.

dlyr committed:
[core][engine] Move triangulate function in Helper namespace.

dlyr committed:
[core][engine] Move triangulate function in Helper namespace.

dlyr committed:
[core] Add a triangulate method that create a triangle layer.

dlyr committed:
[core] Add a triangulate method that create a triangle layer.

dlyr committed:
[core] Add some constness (and const overload) to topomesh.

dlyr committed:
[core] Add some constness (and const overload) to topomesh.

dlyr committed:
[tests] topomesh use makeBox2 (e.g. with MultiIndexed, with Quad).

dlyr committed:
[tests] topomesh use makeBox2 (e.g. with MultiIndexed, with Quad).

dlyr committed:
[core] Remove unneeded inline keyword in topomesh.

dlyr committed:
[core] Remove unneeded inline keyword in topomesh.

dlyr committed:
[examples] Use makeSharpBox2 when appropriate.

dlyr committed:
[examples] Use makeSharpBox2 when appropriate.

dlyr committed:
[engine] Some mesh type update, and sharpBox2.

dlyr committed:
[engine] Some mesh type update, and sharpBox2.

dlyr committed:
[core] Append new layer always unlocked (since a copy).

dlyr committed:
[core] Append new layer always unlocked (since a copy).

dlyr committed:
[gui] Gizmo triangulate before append.

dlyr committed:
[gui] Gizmo triangulate before append.

dlyr committed:
[gui] SimpleWindow, add quit action and connect to shortcut.

dlyr committed:
[gui] SimpleWindow, add quit action and connect to shortcut.

dlyr committed:
[examples] Add quit action to minimalapp.

dlyr committed:
[examples] Add quit action to minimalapp.

dlyr committed:
[tests] topo mesh sharpBox2.

dlyr committed:
[tests] topo mesh sharpBox2.

dlyr committed:
[core] Indexed geom triangulate_any and comments.

dlyr committed:
[core] Indexed geom triangulate_any and comments.

dlyr committed:
[engine] use bracket include.

dlyr committed:
[engine] use bracket include.

dlyr committed:
[gui] Gizmo use triangulate_any.

dlyr committed:
[gui] Gizmo use triangulate_any.

dlyr committed:
[examples] Switch to makeSharpBox2

dlyr committed:
[examples] Switch to makeSharpBox2

dlyr committed:
[engine] VolumeObject Mesh -> GeometryDisplayable.

dlyr committed:
[engine] VolumeObject Mesh -> GeometryDisplayable.

dlyr committed:
[core] Include StdUtils to cout set if needed.

dlyr committed:
[core] Include StdUtils to cout set if needed.

dlyr committed:
[tests] Update index view using MultiIndexedGeometry.

dlyr committed:
[tests] Update index view using MultiIndexedGeometry.

dlyr committed:
[core][engine][...] Remove template from param sphere and torus.

dlyr committed:
[core][engine][...] Remove template from param sphere and torus.

dlyr committed:
[core] Move makeParametricShpere (and Torus) implem to .cpp.

dlyr committed:
[core] Move makeParametricShpere (and Torus) implem to .cpp.

dlyr committed:
[engine] iwyu (since core implem move to .cpp).

dlyr committed:
[engine] iwyu (since core implem move to .cpp).

dlyr committed:
[core] use MultiIndexGeometry for makeGrid.

dlyr committed:
[core] use MultiIndexGeometry for makeGrid.

dlyr committed:
[core] Remove old make*Box.

dlyr committed:
[core] Remove old make*Box.

dlyr committed:
[core] Rename Box2 -> Box.

dlyr committed:
[core] Rename Box2 -> Box.

dlyr committed:
[engine][gui][...] Rename Box2 -> Box.

dlyr committed:
[engine][gui][...] Rename Box2 -> Box.

dlyr committed:
[core] makePlaneGrid returns MultiIndexGeometry.

dlyr committed:
[core] makePlaneGrid returns MultiIndexGeometry.

dlyr committed:
[examples] TriMesh -> MultiIndexedGeometry.

dlyr committed:
[examples] TriMesh -> MultiIndexedGeometry.

dlyr committed:
[engine] Continue update, remove some LineMesh.

dlyr committed:
[engine] Continue update, remove some LineMesh.

dlyr committed:
[examples] PolyMesh -> MultiIndexed.

dlyr committed:
[examples] PolyMesh -> MultiIndexed.

dlyr committed:
[engine] LineMesh -> MultiIndexed.

dlyr committed:
[engine] LineMesh -> MultiIndexed.

dlyr committed:
[engine] DrawPrimitives, remove ref to TriangleMesh and LineMesh.

dlyr committed:
[engine] DrawPrimitives, remove ref to TriangleMesh and LineMesh.

dlyr committed:
[engine] Use MultiIndexed for bone shape.

dlyr committed:
[engine] Use MultiIndexed for bone shape.

dlyr committed:
[examples] Load directly geom from geomData.

dlyr committed:
[examples] Load directly geom from geomData.

dlyr committed:
[engine] Remove templated SurfaceMeshComponentBase.

dlyr committed:
[engine] Remove templated SurfaceMeshComponentBase.

dlyr committed:
[engine] Mesh remove deprecated General,Poly,Triangle,Quad -Mesh.

dlyr committed:
[engine] Mesh remove deprecated General,Poly,Triangle,Quad -Mesh.

dlyr committed:
[engine] Remove deprecated, refactor inline ...

dlyr committed:
[engine] Remove deprecated, refactor inline ...

dlyr committed:
[engine] Remove deprecated Mesh.

dlyr committed:
[engine] Remove deprecated Mesh.

dlyr committed:
[core] Change ref to TriangleMesh:: by MultiIndexedGeometry or Index

dlyr committed:
[core] Change ref to TriangleMesh:: by MultiIndexedGeometry or Index

dlyr committed:
[core] Remove ref to deprecated.

dlyr committed:
[core] Remove ref to deprecated.

dlyr committed:
[tests] Get rid of lasts TriangleMesh in tests.

dlyr committed:
[tests] Get rid of lasts TriangleMesh in tests.

dlyr committed:
[core] Replace TriangleMesh by MultiIndexed equiv.

dlyr committed:
[core] Replace TriangleMesh by MultiIndexed equiv.

dlyr committed:
[core] Refactor semantics, use std helper in IndexedGeometry.

dlyr committed:
[core] Refactor semantics, use std helper in IndexedGeometry.

dlyr committed:
[*] Some comments, revome deprecadted point cloud, /*! -> /**.

dlyr committed:
[*] Some comments, revome deprecadted point cloud, /*! -> /**.

dlyr committed:
[core] Remove conversion ctor

Since not used and error prone, if needed add back as explicit.
Also remove redundent typename keyword.

dlyr committed:
[core] Remove conversion ctor

Since not used and error prone, if needed add back as explicit.
Also remove redundent typename keyword.

dlyr committed:
[core] Remove using redefinition in derived class.

dlyr committed:
[core] Remove using redefinition in derived class.

dlyr committed:
[core] Remove redundant inline in indexed geometry.

dlyr committed:
[core] Remove redundant inline in indexed geometry.

dlyr committed:
[engine] Add missing SurfaceMeshComponent ctor.

dlyr committed:
[core] Add missing SurfaceMeshComponent ctor.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

dlyr committed:
fix

dlyr committed:
[gui] Add missing include.

dlyr committed:
[io] Recenter point cloud data (wip, need to add control on this).
